sammy/sammy.py

```diff
@@ -161,6 +161,58 @@ def complete_update():
   """
   logging.info("Not implemented.")
 
+@platform.command()
+def flash():
+  def stop_flash(reason, prompt_clone_or_setup=False):
+    logging.error(reason)
+    if prompt_clone_or_setup:
+      logging.error(f'Please clone the SJSU-Dev2 repository or rerun ./setup')
+    sys.exit(1)
+
+  def get_tools_dir():
+    SJSU_MK_PATH = os.path.expanduser('~') + '/.sjsu_dev2.mk'
+    base_path = None
+    if not os.path.exists(SJSU_MK_PATH):
+      stop_flash('File ~/.sjsu_dev2.mk not found!', prompt_clone_or_setup=True)
+    with open(SJSU_MK_PATH) as file:
+      lines = file.readlines()
+      for line in lines:
+        if 'SJSU_DEV2_BASE' in line:
+          base_path = line.split('= ')[-1].strip()
+          break
+    if base_path == None or not os.path.isdir(base_path):
+      stop_flash('Could not resolve SJSU-Dev2 base path!',
+                 prompt_clone_or_setup=True)
+    else:
+      return base_path + '/tools'
+
+  def get_binary_path():
+    WORKING_DIRECTORY = os.getcwd() + '/'
+    # ensure we are in dev2
+    if 'SJSU-Dev2' not in WORKING_DIRECTORY:
+      stop_flash(
+          'Please run the flash command from within the SJSU-Dev2 directory.')
+    directory_contents = os.listdir()
+    if 'build' not in directory_contents:
+      stop_flash(
+          'Please run the flash command from within your project directory.')
+    os.chdir('build')
+    directory_contents = list(os.walk('.'))
```

Comment:
os.walk returns a result like below, and i iterate over each element to find the `firmware.bin`.
```py
[
  ('.', ['lpc40xx'], []),
  ('./lpc40xx', ['g'], []),
  ('./lpc40xx/g', ['objects'], ['firmware.hex', 'firmware.elf', 'firmware.map', 'firmware.siz', 'firmware.bin', 'firmware.lst', 'firmware.src.lst']),
  ('./lpc40xx/g/objects', ['source', 'g'], []), ('./lpc40xx/g/objects/source', [], ['main.cpp.d', 'main.cpp.o']), 
  ('./lpc40xx/g/objects/g', [], ['libsjsudev2.a'])
]
```